pkg/scheduler/framework/runtime/framework.go

```diff
@@ -653,11 +653,11 @@ func (f *frameworkImpl) RunPreFilterPlugins(ctx context.Context, state *framewor
 			continue
 		}
 		if !s.IsSuccess() {
-			s.SetFailedPlugin(pl.Name())
-			if s.IsUnschedulable() {
-				return nil, s
+			if s.IsError() {
+				return nil, framework.AsStatus(fmt.Errorf("running PreFilter plugin %q: %w", pl.Name(), s.AsError()))
 			}
-			return nil, framework.AsStatus(fmt.Errorf("running PreFilter plugin %q: %w", pl.Name(), s.AsError())).WithFailedPlugin(pl.Name())
+			s.SetFailedPlugin(pl.Name())
+			return nil, s
 		}
```

Comment:
So far, how to treat the plugin returning error is different in each extension point. This PR unifies it; the plugin returning error won't be registered in failed plugins.